utils/logging_utils.py

```diff
@@ -109,14 +110,42 @@ def add_text(self, tag, text, step=None):
                             step=step)
     
     @ignore_exception
-    def display_attention_heads(self, outputs, tag='', step=None):
+    def display_attention_heads(self, outputs, tag='', step=None, fname=None):
```

Comment:
Type hinting would be really helpful here.

Reply:
```suggestion
    def display_attention_heads(self, outputs: dict, tag='', step=None, fname=None):
```